src/sqlfluff/rules/L016.py

```diff
@@ -500,7 +518,10 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                         idx -= 1
                     else:
                         break  # pragma: no cover
-                create_elements = line_indent + [this_line[-1], context.segment]
+                create_elements = line_indent + [
+                    this_line[-1],
+                    cast(RawSegment, context.segment),
```

Comment:
No change in functionality. mypy was complaining about `RawSegment` vs `BaseSegment` here, and adding the cast made it happy.